test/drivers/test_rbki.cc

```diff
@@ -0,0 +1,125 @@
+            row = m;
+            col = n;
+        }
+    };
+
+
+    // This routine computes the residual norm error, consisting of two parts (one of which) vanishes
+    // in exact precision. Target_rank defines size of U, V as returned by RBKI; custom_rank <= target_rank.
+    template <typename T>
+    static T
+    residual_error_comp(RBKITestData<T> &all_data, int64_t custom_rank) {
+        auto m = all_data.row;
+        auto n = all_data.col;
+
+        T* U_cpy_dat = RandLAPACK::util::upsize(m * n, all_data.U_cpy);
+        T* VT_cpy_dat = RandLAPACK::util::upsize(n * n, all_data.VT_cpy);
+
+        lapack::lacpy(MatrixType::General, m, n, all_data.U.data(), m, U_cpy_dat, m);
+        lapack::lacpy(MatrixType::General, n, n, all_data.VT.data(), n, VT_cpy_dat, n);
+
+        // AV - US
+        // Scale columns of U by S
+        for (int i = 0; i < custom_rank; ++i)
+            blas::scal(m, all_data.Sigma[i], &U_cpy_dat[m * i], 1);
+
+
+        // Compute AV(:, 1:custom_rank) - SU(1:custom_rank)
+        blas::gemm(Layout::ColMajor, Op::NoTrans, Op::Trans, m, custom_rank, n, 1.0, all_data.A.data(), m, all_data.VT.data(), n, -1.0, U_cpy_dat, m);
+
+
+        // A'U - VS
+        // Scale columns of V by S
+        // Since we have VT, we will be scaling its rows
+        // The data is, however, stored in a column-major format, so it is a bit weird.
+        //for (int i = 0; i < n; ++i)
+        //    blas::scal(custom_rank, all_data.Sigma[i], &VT_cpy_dat[i], n);
+        for (int i = 0; i < custom_rank; ++i)
+            blas::scal(n, all_data.Sigma[i], &VT_cpy_dat[i], n);
+        // Compute A'U(:, 1:custom_rank) - VS(1:custom_rank).
+        // We will actually have to perform U' * A - Sigma * VT.
+
+        blas::gemm(Layout::ColMajor, Op::Trans, Op::NoTrans, custom_rank, n, m, 1.0, all_data.U.data(), m, all_data.A.data(), m, -1.0, VT_cpy_dat, n);
+
+        T nrm1 = lapack::lange(Norm::Fro, m, custom_rank, U_cpy_dat, m);
+        T nrm2 = lapack::lange(Norm::Fro, custom_rank, n, VT_cpy_dat, n);
+
+        return std::sqrt(std::pow(nrm1, 2) + std::pow(nrm2, 2));
+    }
+
+
+    template <typename T, typename RNG, typename alg_type>
+    static void test_RBKI_general(
+        int64_t b_sz,
+        int64_t target_rank,
+        int64_t custom_rank,
+        RBKITestData<T> &all_data,
+        alg_type &RBKI,
+        RandBLAS::RNGState<RNG> &state) {
+
+        auto m = all_data.row;
+        auto n = all_data.col;
+        RBKI.max_krylov_iters = (int) ((target_rank * 2) / b_sz);
+
+        RBKI.call(m, n, all_data.A.data(), m, b_sz, all_data.U.data(), all_data.VT.data(), all_data.Sigma.data(), state);
+        // Compute singular values via a deterministic method
+
+        T residual_err_custom = residual_error_comp<T>(all_data, custom_rank);
+        printf("residual_err_custom %e\n", residual_err_custom);
+        ASSERT_NEAR(residual_err_custom, 8.039386e-13, std::pow(std::numeric_limits<T>::epsilon(), 0.825));
```

Comment:
The reference value here is specific to double precision. Since you're expecting that something is basically numerically zero, it's better to use ``ASSERT_LE`` (for less than or equal to), a-la
```c++
ASSERT_LE(residual_err_custom, std::pow(std::numeric_limits<T>::epsilon(), 0.825));
```

Reply:
Fixed.